components/editor/DragItemWithDropZone.tsx

```diff
@@ -60,23 +62,79 @@ const DragItemWithDropZone: FC<DragItemWithDropZoneProps> = ({
     [greedy, setHasDropped, setHasDroppedOnChild]
   )
 
+  const handleDownClick = (id: any) => {
+    dispatch({
+      payload: {
+        direction: 'down',
+        id,
+      },
+      type: 'MOVE_ITEM',
+    })
+  }
+
+  const handleUpClick = (id: any) => {
+    dispatch({
+      payload: {
+        direction: 'up',
+        id,
+      },
+      type: 'MOVE_ITEM',
+    })
+  }
+
   return (
     <div ref={drag}>
       <div ref={drop}>
         {(() => {
           switch (type) {
             case PrimitiveTypes.PRIMITIVE_CONTAINER:
```

Comment:
Do we need to add a case for every primitive block we offer?

Reply:
I don't think so, we can consolidate them for now.